lgn-provers/src/params/mod.rs

```diff
@@ -18,120 +16,46 @@ use checksums::ops::write_hash_comparison_results;
 use tracing::debug;
 use tracing::error;
 use tracing::info;
+use tracing::warn;
 
 pub struct ParamsLoader;
 
 // Could make configurable but 3600 should be enough
 const HTTP_TIMEOUT: u64 = 3600;
 const DOWNLOAD_MAX_RETRIES: u8 = 3;
 
-impl ParamsLoader
+/// Utility to open a file and format an error if it errors.
+///
+/// [std::fs::File::open] does not format the file name into its error, it only returns NotFound.
+fn open(file_path: &PathBuf) -> anyhow::Result<File>
 {
-    pub fn prepare_bincode<P: for<'a> serde::de::Deserialize<'a>>(
```

Comment:
prepare_raw does the same, but doesn't decode using bincode. I moved the decoding to the caller and remove this function.